app/javascript/components/misc/CodeMirror/languageCompartment.ts

```diff
@@ -7,177 +7,227 @@ export const loadLanguageCompartment = async (
   language: string
 ): Promise<Extension> => {
   switch (language) {
-    case 'cpp':
+    case 'cpp': {
       const { cpp } = await import('@codemirror/lang-cpp')
       return compartment.of(cpp())
-    case 'java':
+    }
+    case 'java': {
       const { java } = await import('@codemirror/lang-java')
       return compartment.of(java())
-    case 'cfml':
+    }
+    case 'cfml': {
       const { javascript: cfml } = await import('@codemirror/lang-javascript')
       return compartment.of(cfml())
+    }
     case 'javascript':
-    case 'typescript':
+    case 'typescript': {
```

Comment:
The documentation's "When not to use" section says:

> If you depend on fall through behavior and want access to bindings introduced in the case block.

Should we copy the code to not have fall through behavior then?

Reply:
No, this works.

```typescript
switch (language) {
  case 'javascript':
  case 'typescript': { 
    // runs on both js and ts
    console.log(language);
  } 
}

// => 'javascript' is logged
```